### PR TITLE
Ft/160451507/ Users should recieve In-app notifications

### PR DESCRIPTION
**Pull request by udochukwu**
**What does this PR do?**

- Sends In-app notifications to users

**Description of Task to be completed?**

- Setup a notification controller for sending notifications

- Add more  methods to the notification controller

- Refactor the notification controller

**What are the relevant pivotal tracker stories?**

https://www.pivotaltracker.com/story/show/160451507

**Screenshots (if appropriate)**

- N/A

**Questions**

- N/A

### CODE REVIEW

File: server/models/user.js [hunk 1]
```diff
@@ -67,6 +67,11 @@ export default (sequelize, DataTypes) => {
     verifyToken: {
       type: DataTypes.STRING
     },
+    emailNotification: {
+      allowNull: false,
+      type: DataTypes.BOOLEAN,
+      defaultValue: true
+    },
```
Comment (udochukwu) [inline server/models/user.js]: This new field checks if a user opted in/out of email notifications.

**Review by slimsolz — COMMENTED** [review]
LGTM

**Review by onwuvic — COMMENTED** [review]
LGTM

**Review by samdiano — COMMENTED** [review]
LGTM

**Review by s-akinrele — APPROVED** [review]

**Comment by onwuvic** [comment]
Why is your test failing?

**Pull request merged**